File: ftgo-kitchen-service/src/main/java/me/jangjunha/ftgo/kitchen_service/grpc/ClientCallCredentials.java
package me.jangjunha.ftgo.kitchen_service.grpc;

import me.jangjunha.ftgo.common.auth.ExplicitCallCredentials;

public class ClientCallCredentials extends ExplicitCallCredentials {

    public ClientCallCredentials() {
        super("x-ftgo-authenticated-client-id", "ftgo-kitchen-service");
    }
}

File: ftgo-api-gateway/src/main/kotlin/me/jangjunha/ftgo/apigateway/security/grpc/ClientCallCredentials.kt
package me.jangjunha.ftgo.apigateway.security.grpc

import io.grpc.CallCredentials
import io.grpc.Metadata
import java.util.concurrent.Executor

object ClientCallCredentials : CallCredentials() {
    override fun applyRequestMetadata(requestInfo: RequestInfo, appExecutor: Executor, applier: MetadataApplier) {
        appExecutor.execute {
            applier.apply(Metadata().apply {
                put(
                    Metadata.Key.of("x-ftgo-authenticated-client-id", Metadata.ASCII_STRING_MARSHALLER),
                    "ftgo-api-gateway"
                )
            })
        }
    }
}
